### PR TITLE
HPCC-397 Tidy up hthor/roxie input interfaces

### PR DESCRIPTION
**Pull request by richardkchapman**
Initial steps towards merging roxie/thor rowstream interfaces

### CODE REVIEW

File: roxie/ccd/ccdserver.cpp [hunk 1]
```diff
@@ -19358,10 +19369,10 @@ class CRoxieServerWhenActivity : public CRoxieServerActivity
         }
         else if (state != STATEstopped)
         {
-            stopDependencies(savedExtractSize, savedExtract, aborting ? WhenSuccessId : WhenFailureId);  // These ones don't get executed
-            executeDependencies(savedExtractSize, savedExtract, aborting ? WhenFailureId : WhenSuccessId); // These ones do
+            stopDependencies(savedExtractSize, savedExtract, aborted ? WhenSuccessId : WhenFailureId);  // These ones don't get executed
+            executeDependencies(savedExtractSize, savedExtract, aborted ? WhenFailureId : WhenSuccessId); // These ones do
```
Comment (ghalliday) [inline roxie/ccd/ccdserver.cpp]: I'm not sure this is right.   This activity is not a sink, so I don't think aborted will be called.  This should probably catch exception inside nextOnGroup() and use that to set a flag if it failed.

File: roxie/ccd/ccdserver.cpp [hunk 2]
```diff
@@ -1203,11 +1206,18 @@ class CRoxieServerActivity : public CInterface, implements IRoxieServerActivity,
...
+                    input->stop();
             }
         }
     }
 
+    virtual void abort()
+    {
+        aborted = true;
+        // MORE - should abort dependencies here?
+        stop();
```
Comment (ghalliday) [inline roxie/ccd/ccdserver.cpp]: I am not sure if abort() should call stop(), or whether they should be separate.

E.g., you may want to call abort() on all activities when a process is guillotined.

Comment (richardkchapman) [inline roxie/ccd/ccdserver.cpp]: I still want to stop them... calling abort first means that the sink at least knows not to do any extra work.

File: roxie/ccd/ccdserver.cpp (hunk 1 above)
Comment (richardkchapman) [inline roxie/ccd/ccdserver.cpp]: alternatively the graph::abort code should set aborted on all activities not just the sinks

**Comment by richardkchapman** [comment]
@ghalliday Please review. If you are happy I will squash down to 2 commits before you merge...

**Pull request merged**